tests/test_scalars.py

```diff
@@ -15,26 +22,568 @@
+    check(assert_type(ts.asm8, np.datetime64), np.datetime64)
+    check(assert_type(ts.day_of_week, int), int)
+    check(assert_type(ts.day_of_year, int), int)
+    check(assert_type(ts.dayofweek, int), int)
+    check(assert_type(ts.dayofyear, int), int)
+    check(assert_type(ts.days_in_month, int), int)
+    check(assert_type(ts.daysinmonth, int), int)
+    check(assert_type(ts.is_leap_year, bool), bool)
+    check(assert_type(ts.is_month_end, bool), bool)
+    check(assert_type(ts.is_month_start, bool), bool)
+    check(assert_type(ts.is_quarter_end, bool), bool)
+    check(assert_type(ts.is_quarter_start, bool), bool)
+    check(assert_type(ts.is_year_end, bool), bool)
+    check(assert_type(ts.is_year_start, bool), bool)
+    check(assert_type(ts.quarter, int), int)
+    check(assert_type(ts.tz, Optional[dt.tzinfo]), type(None))
+    check(assert_type(ts.week, int), int)
+    check(assert_type(ts.weekofyear, int), int)
+    check(assert_type(ts.day, int), int)
+    check(assert_type(ts.fold, int), int)
+    check(assert_type(ts.hour, int), int)
+    check(assert_type(ts.microsecond, int), int)
+    check(assert_type(ts.minute, int), int)
+    check(assert_type(ts.month, int), int)
+    check(assert_type(ts.nanosecond, int), int)
+    check(assert_type(ts.second, int), int)
+    check(assert_type(ts.tzinfo, Optional[dt.tzinfo]), type(None))
+    check(assert_type(ts.value, int), int)
+    check(assert_type(ts.year, int), int)
+
+
+def test_timestamp_add_sub() -> None:
+    ts = pd.Timestamp("2000-1-1")
+    np_td64_arr: npt.NDArray[np.timedelta64] = np.array([1, 2], dtype="timedelta64[ns]")
+
+    as_pd_timedelta = pd.Timedelta(days=1)
+    as_dt_timedelta = dt.timedelta(days=1)
+    as_offset = 3 * Day()
+    as_timedelta_index = pd.TimedeltaIndex([1, 2, 3], "D")
+    as_timedelta_series = pd.Series(as_timedelta_index)
+    check(assert_type(as_timedelta_series, TimedeltaSeries), pd.Series)
+    as_np_ndarray_td64 = np_td64_arr
+
+    check(assert_type(ts + as_pd_timedelta, pd.Timestamp), pd.Timestamp)
+    check(assert_type(as_pd_timedelta + ts, pd.Timestamp), pd.Timestamp)
+
+    check(assert_type(ts + as_dt_timedelta, pd.Timestamp), pd.Timestamp)
+    check(assert_type(as_dt_timedelta + ts, pd.Timestamp), pd.Timestamp)
+
+    check(assert_type(ts + as_offset, pd.Timestamp), pd.Timestamp)
+    check(assert_type(as_offset + ts, pd.Timestamp), pd.Timestamp)
+
+    check(assert_type(ts + as_timedelta_index, pd.DatetimeIndex), pd.DatetimeIndex)
+    check(assert_type(as_timedelta_index + ts, pd.DatetimeIndex), pd.DatetimeIndex)
+
+    check(assert_type(ts + as_timedelta_series, "TimestampSeries"), pd.Series)
+    check(assert_type(as_timedelta_series + ts, "TimestampSeries"), pd.Series)
+
+    check(assert_type(ts + as_np_ndarray_td64, npt.NDArray[np.datetime64]), np.ndarray)
+    # pyright and mypy disagree on the type of this expression
+    # pyright: Any, mypy: npt.NDArray[np.datetime64]
```

Comment:
this is most likely fixed in latest `pyright` release, so can you remove above comment and `#pyright ignore` and test?

Reply:
It is.  I've bumped pytight to latest too in poetry.